src/search/task_proxy.h

```diff
@@ -558,84 +551,71 @@ class GoalsProxy : public ConditionsProxy {
 
 
 bool does_fire(const EffectProxy &effect, const State &state);
-bool does_fire(const EffectProxy &effect, const GlobalState &state);
 
 
 class State {
     const AbstractTask *task;
-    std::vector<int> values;
+    mutable const StateRegistry *registry;
+    mutable StateID id;
+    mutable const PackedStateBin *buffer;
+    mutable std::shared_ptr<std::vector<int>> values;
```

Comment:
Probably only `values` should be mutable. The other ones are only set in the constructor or assignment operator (which should be non-const).